### PR TITLE
chore: add terraform Appveyor testing jobs

### PR DESCRIPTION
**Pull request by moelasmar**
#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write/update unit tests
- [ ] Write/update integration tests
- [ ] Write/update functional tests if needed
- [ ] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/hook_packages/terraform/hooks/prepare/resource_linking.py [hunk 1]
```diff
@@ -640,10 +640,25 @@ def _process_resolved_layers(
         "Map the resolved layers %s to configuration function %s.", resolved_layers, function_tf_resource.full_address
     )
     layers = []
+    does_refer_to_constant_values = False
+    does_refer_to_data_sources = False
     for resolved_layer in resolved_layers:
         # Skip ConstantValue layers reference, as it will be already handled by terraform plan command.
-        if isinstance(resolved_layer, ResolvedReference):
-            layers += _process_reference_layer_value(function_tf_resource, resolved_layer, tf_layers)
+        if isinstance(resolved_layer, ConstantValue):
```
Comment (sriram-mv) [inline samcli/hook_packages/terraform/hooks/prepare/resource_linking.py:647]: This is already done in another PR right? it could be removed.

File: appveyor-ubuntu.yml [hunk 2]
```diff
@@ -184,12 +189,82 @@ for:
...
+      only:
+        - image: Ubuntu
+          configuration: LocalZipTerraformBuildIntegTesting
+
+    test_script:
+      # Pre-dev Tests
+      - "pip install -e \".[pre-dev]\""
+
+      - "pip install -e \".[dev]\""
+      - sh: "pytest -vv -n 4 tests/integration/buildcmd/test_build_terraform_applications.py::TestBuildTerraformApplicationsWithZipBasedLambdaFunctionAndLocalBackend_0 --json-report --json-report-file=TEST_REPORT-integration-buildcmd.json"
```
Comment (sriram-mv) [inline appveyor-ubuntu.yml:210]: I like the `-n 4` there is also an auto variable you could use on this. Can this level of parallelization cause flaky tests?

**Review by sriram-mv — COMMENTED** [review]
What does overall run time look like?

File: samcli/hook_packages/terraform/hooks/prepare/resource_linking.py (hunk 1 above)
Comment (moelasmar) [reply samcli/hook_packages/terraform/hooks/prepare/resource_linking.py:647]: yes, It seems by mistake I merged this change https://github.com/aws/aws-sam-cli/pull/4387 to this one before I create the PR :(

**Review by sriram-mv — APPROVED** [review]
Approving this with comments.

**Review by hawflau — APPROVED** [review]
